### ISSUE TITLE
Textarea border is partially cut by the floating label

### ISSUE THREAD

**Issue by ricardobrandao** [material-components/material-components-web#1608]
### What MDC-Web Version are you using?

> 0.25.0 (except for base that is 0.24.0

### What browser(s) is this bug affecting?

> Chrome 62.0.3202.94 (64 bits)

### What OS are you using?

> MacOS 10.13.1

### What are the steps to reproduce the bug?

> Please write the steps which need to be taken in order to reproduce the bug. These steps should be
> as detailed as possible, e.g.
>
> 1. Go to [this url](https://codepen.io/ricardobrandao/pen/xPPymW).
> 2. Start typing on the textarea
> 3. Observe the component's behavior: the top border is partially overlapped by the label.

### What is the expected behavior?

> The top border should stay the same

### What is the actual behavior?

> The top border is partially cut near the label

### Any other information you believe would be useful?

<img width="291" alt="screen shot 2017-11-17 at 15 37 41" src="https://user-images.githubusercontent.com/2292745/32955384-b8e93692-cbad-11e7-9aeb-ae3220eab6cc.png">

**Comment by amsheehan** [comment]
Looks like a bug. Putting in tracker.

**Comment by mellogarrett** [comment]
Textarea for Safari (Mobile & Desktop) is even worse ☹️ 

<img width="746" alt="screen shot 2017-12-04 at 12 35 25 pm" src="https://user-images.githubusercontent.com/9908831/33566985-b56b68b0-d8ef-11e7-8a07-a0a08b7f8bf1.png">

**Comment by amsheehan** [comment]
We have found that zoom levels and pixel densities of monitors have had a detrimental effect on these types of features. @mellogarrett Thank you for the screen shot, could you also provide the specs for the machine(s) you're using?

**Comment by mellogarrett** [comment]
Seen on both 2017 MacBook Pro & iPhone 8+

**Comment by moog16** [comment]
While looking at the dev tools, it seems like the <textarea /> and the `mdc-text-field` wrapper both have a border. The <textarea /> element does not have border-radius, and since the label is position: absolute, it overlaps the border. 

Proposed solution is to remove the textarea border. Another is to remove all the classes from the mdc-text-field, but it then not be consistent with the text field.

**Comment by moog16** [comment]
@ricardobrandao In the MDC demo normalize.css is being used, but is turned off in your codepen. I noticed that margin is added by default on both Chrome and Safari. Safari (v11.0.1) however adds by default to the textfield.

```
margin-top: 0em;
margin-bottom: 0em;
margin-left: 0em;
margin-right: 0em;
```

I noticed that this doesn't remove all margin, but by overriding this with `margin: 0` Safari actually removes the margin as expected. So seems like we stumbled upon a Safari bug. I'll be opening a PR with that change.
